pkg/apply/prune/prune.go

```diff
@@ -200,58 +198,18 @@ func (p *Pruner) removeInventoryAnnotation(obj *unstructured.Unstructured) (*uns
 	// This prevents race conditions when writing to the underlying map.
 	obj = obj.DeepCopy()
 	id := object.UnstructuredToObjMetadata(obj)
-	annotations := obj.GetAnnotations()
-	if annotations != nil {
-		if _, ok := annotations[inventory.OwningInventoryKey]; ok {
-			klog.V(4).Infof("removing annotation (object: %q, annotation: %q)", id, inventory.OwningInventoryKey)
-			delete(annotations, inventory.OwningInventoryKey)
-			obj.SetAnnotations(annotations)
-			namespacedClient, err := p.namespacedClient(id)
-			if err != nil {
-				return obj, err
-			}
-			_, err = namespacedClient.Update(context.TODO(), obj, metav1.UpdateOptions{})
+	if inventory.OwningInventoryAnnotation(obj) != "" {
```

Comment:
The addition and use of the OwningInventoryAnnotation and DeleteOwningInventoryAnnotation functions could be moved to its own PR. It's just a cleanup, not dependent on the inv client changes.